### PR TITLE
Add data_collator with attention_mask feature

### PR DESCRIPTION
**Pull request by LunaBlack**
For models like BERT, we should mask the padding ids as invalidate to avoid attention on them. Therefore, in addition to `input_ids`, the feature `attention_mask` should also be fed to the model. 
Add a data collator `DataCollatorForMaskedLanguageModeling` which will return all of above data, and add a dataset class `LineByLineTextMaskDataset` which is related with the former.

### CODE REVIEW

**Comment by LunaBlack** [comment]
Add a related issue: [https://github.com/huggingface/transformers/issues/4702](url)

**Comment by LysandreJik** [comment]
I agree that we should handle `attention_mask`, but I don't think adding another class `DataCollatorForMaskedLanguageModeling` is necessary. The class `DataCollatorForLanguageModeling` already handles masked language modeling, and should imo be modified to handle attention masks as well.

**Comment by LunaBlack** [comment]
@LysandreJik 
I agree!
I have modified the code, and passed related unit tests. 
Could you please help review the code?
